cmd/entrypoint-wrapper/main.go

```diff
@@ -64,28 +70,31 @@ func main() {
 }
 
 type options struct {
-	dry            bool
-	name           string
-	srcPath        string
-	dstPath        string
-	waitPath       string
-	waitTimeoutStr string
-	waitTimeout    time.Duration
-	skipKubeconfig bool
-	cmd            []string
-	client         coreclientset.SecretInterface
+	dry              bool
+	name             string
+	srcPath          string
+	dstPath          string
+	waitPath         string
+	waitTimeoutStr   string
+	waitTimeout      time.Duration
+	mode             string
+	rwKubeconfig     bool
+	uploadKubeconfig bool
+	updateSharedDir  bool
+	cmd              []string
+	client           coreclientset.SecretInterface
 }
 
 func bindOptions(flag *flag.FlagSet) *options {
 	opt := &options{}
 	flag.BoolVar(&opt.dry, "dry-run", false, "Print the secret instead of creating it")
 	flag.StringVar(&opt.waitPath, "wait-for-file", "", "Wait for a file to appear at this path before starting the program")
 	flag.StringVar(&opt.waitTimeoutStr, "wait-timeout", "", "Used with --wait-for-file, maximum wait time before starting the program")
-	flag.BoolVar(&opt.skipKubeconfig, "skip-kubeconfig", false, "Skip managing the $KUBECONFIG copy")
+	flag.StringVar(&opt.mode, "mode", manageKubeconfigMode, fmt.Sprintf("Set how kubeconfig should be managed. Allowed values are: %s, %s or %s", manageKubeconfigMode, skipKubeconfigMode, observerMode))
```

Comment:
will that break Steve's feature? Probably he would need to update his configs.

Reply:
What is the default, if you don't provide anything `manageKubeconfigMode` is set? It should.

Reply:
It should not break Steve's work as I have translated his `--skip-kubeconfig` flag into `--mode=skip-kubeconfig` with no changes in behaviour: to dig deeper take a look here #2930.
If `--mode` was not specified then the default value is `manageKubeconfigMode` that is `"manage-kubeconfig"`. The third parameter of `flag.StringVar` method is the default value.